src/Interpreters/DatabaseCatalog.cpp

```diff
@@ -291,7 +342,37 @@ DatabaseAndTable DatabaseCatalog::getTableImpl(
         {
             assert(!db_and_table.first && !db_and_table.second);
             if (exception)
-                exception->emplace(Exception(ErrorCodes::UNKNOWN_TABLE, "Table {} doesn't exist", table_id.getNameForLogs()));
+            {
+                if (!db_and_table.first)
+                {
+                    DatabaseNameHints hints(*this);
+                    std::vector<String> names = hints.getHints(table_id.getDatabaseName(), hints.getAllRegisteredNames());
```

Comment:
...and all such calls below
```suggestion
                    std::vector<String> names = hints.getHints(table_id.getDatabaseName());
```